libraries/chain/include/eosio/chain/wasm_interface.hpp

```diff
@@ -20,63 +20,13 @@ namespace eosio { namespace chain {
       int32_t code = 0;
    };
 
-   namespace webassembly { namespace common {
-      class intrinsics_accessor;
-
-      class root_resolver : public Runtime::Resolver {
-      public:
-         // The non-default constructor puts root_resolver in a mode where it does validation, i.e. only allows "env" imports.
-         // This mode is used by the generic validating code that runs during setcode, where we only want "env" to pass.
-         // The default constructor is used when no validation is required such as when the wavm runtime needs to
-         // allow linkage to the intrinsics and the injected functions.
-
-         root_resolver() {}
-
-         root_resolver( const whitelisted_intrinsics_type& whitelisted_intrinsics )
-         :whitelisted_intrinsics(&whitelisted_intrinsics)
-         {}
-
-         bool resolve(const string& mod_name,
-                      const string& export_name,
-                      IR::ObjectType type,
-                      Runtime::ObjectInstance*& out) override
-         { try {
-            bool fail = false;
-
-            if( whitelisted_intrinsics != nullptr ) {
-               // Protect access to "private" injected functions; so for now just simply allow "env" since injected
-               // functions are in a different module.
-               EOS_ASSERT( mod_name == "env", wasm_exception,
```

Comment:
- If the name is not whitelisted, it will be rejected by the whitelist check
- If it is whitelisted, it will be rejected by the link check because the imports available in "env" are disjoint from the imports available in eosio_injected_module_name.

I'll add an explicit check for env in eos-vm.cpp to make it more future proof.